src/core/transaction.rs

```diff
@@ -444,17 +997,58 @@ impl<'a> Clone for TxOutRef<'a> {
 impl<'a> Copy for TxOutRef<'a> {}
 
 /// Common operations for transaction inputs, implemented by both owned and borrowed types.
+///
+/// This trait provides shared functionality for [`TxIn`] and [`TxInRef`],
+/// allowing code to work with either owned or borrowed inputs.
+///
+/// # Examples
+///
+/// ```no_run
+/// use bitcoinkernel::{prelude::*, Transaction};
+///
+/// fn print_outpoint<T: TxInExt>(input: &T) {
+///     let outpoint = input.outpoint();
+///     println!("Spending {}:{}", outpoint.txid(), outpoint.index());
+/// }
+/// ```
 pub trait TxInExt: AsPtr<btck_TransactionInput> {
     /// Returns a reference to the outpoint being spent by this input.
     ///
-    /// The outpoint identifies which previous transaction output this input is spending.
+    /// The outpoint identifies which previous transaction output this input is spending
+    /// by referencing the transaction ID and output index.
+    ///
+    /// # Examples
+    /// ```no_run
+    /// # use bitcoinkernel::{prelude::*, Transaction, KernelError};
+    /// # fn example(tx: &Transaction) -> Result<(), KernelError> {
+    /// let input = tx.input(0)?;
+    /// let outpoint = input.outpoint();
+    ///
+    /// println!("Spending output {} from transaction {}",
+    ///          outpoint.index(),
+    ///          outpoint.txid());
+    /// # Ok(())
+    /// # }
+    /// ```
     fn outpoint(&self) -> TxOutPointRef<'_> {
         let ptr = unsafe { btck_transaction_input_get_out_point(self.as_ptr()) };
         unsafe { TxOutPointRef::from_ptr(ptr) }
     }
 }
 
 /// A single transaction input referencing a previous output to be spent.
+///
+/// Transaction inputs specify which previous outputs are being spent. Each input contains:
+/// - **Outpoint**: Reference to the previous output (txid + index)
+/// - **Script signature**: Proof that the spender can satisfy the output's conditions
+/// - **Sequence**: Used for relative lock time and replace-by-fee
+///
+/// **Note**: The script signature and sequence fields are not currently accessible
+/// through this API. You can access the outpoint via [`outpoint`](TxInExt::outpoint).
```

Comment:
I think we should just skip all of this. We can always add a complete description once all the getters are there.